indico/modules/receipts/client/js/printing/PrintReceiptsModal.jsx

```diff
@@ -82,13 +83,30 @@ const archiveFilename = {
 export default function PrintReceiptsModal({onClose, registrationIds, eventId}) {
   const [receiptIds, setReceiptIds] = useState([]);
   const [downloading, setDownloading] = useState(false);
+  const [eventImages, setEventImages] = useState([]);
+  const [loadingImages, setLoadingImages] = useState(false);
 
   const {data: templateList, loading} = useIndicoAxios(allTemplatesURL({event_id: eventId}), {
     trigger: eventId,
     camelize: true,
   });
   const ready = !loading && !!templateList;
 
+  const fetchImages = async () => {
+    if (loadingImages) {
+      return;
+    }
+    setLoadingImages(true);
+    let response;
+    try {
+      response = await indicoAxios.get(eventImagesURL({event_id: eventId}));
+      setEventImages(response.data.images);
+    } catch (error) {
+      return handleAxiosError(error);
+    }
+    setLoadingImages(false);
+  };
```

Comment:
You can most likely replace this whole thing with `useIndicoAxios()` so it runs only once (immediately when opening the dialog)

Reply:
apparently that solved the whole update-preview-on-refresh issue 😄 
I guess because now i'm keeping the state inside the `TemplateParameterEditor`, so now it doesn't need to update the whole `PrintReceiptsModal`